webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -152,25 +162,116 @@ def update_correlation_plot(
+
+        def disable_figure() -> Tuple[Dict, go.Figure, Dict, Dict]:
+            parent.save_state("correlation-store-obs-range", {})
+            return {"display": "none", "min-height": "50px"}, go.Figure(), {}, {}
+
+        ensembles = _get_selected_ensembles_from_store(ensemble_selection_store)
+        if not (ensembles and responses):
+            return disable_figure()
+        if triggered_id == parent.uuid("obs_index_selector") and selected_data is None:
+            store_obs_range = {}
+
+        if selected_data is None or "range" not in selected_data:
+            selected_data = {"range": store_obs_range}
+
+        # Use only one ensemble
+        ensemble = load_ensemble(parent, ensembles[0])
+        obs_plots: List[PlotModel] = []
+        index_axis = False
+        time_axis = False
+        for idx, response_name in enumerate(responses):
+            response = ensemble.responses[response_name]
+            if response.observations:
+                for obs in response.observations:
+                    if not obs.axis:
+                        continue
+                    if obs.axis_type == AxisType.INDEX and not index_axis:
+                        index_axis = True
+                    elif not time_axis:
+                        time_axis = True
+                    obs_plots.append(_observation_index_plot(obs, response_name, idx))
+
+        if not obs_plots:
+            return disable_figure()
+
+        fig = go.Figure(layout_yaxis_range=[-1, len(obs_plots)])
+        layout = assets.ERTSTYLE["figure"]["layout"].copy()
+        layout.update(
+            dict(
+                xaxis=XAxis(title="Date", showgrid=False),
+                xaxis2=XAxis(overlaying="x", title="Index", side="top", showgrid=False),
+                dragmode="select",
+                template="plotly_white",
+                height=300,
+            )
+        )
+        fig.update_layout(layout)
+
+        for plot in obs_plots:
+            fig.add_trace(plot.repr)
+
+        store_obs_range = selected_data["range"]
+        selected_indexes = _get_selected_indexes(obs_plots, store_obs_range)
+
+        # Add selection rectangle to figure
+        x_range = store_obs_range.get("x")
+        if not time_axis:
+            x_range = store_obs_range.get("x2")
+
+        if selected_indexes and x_range:
+            x_start, x_end = x_range
+            fig.add_shape(
+                type="rect",
+                yref="paper",
+                x0=x_start,
+                y0=0,
+                x1=x_end,
+                y1=1,
+                fillcolor="PaleTurquoise",
+                opacity=0.35,
+            )
+        else:
+            # Use first index if nothing is selected
+            for p in obs_plots:
+                selected_indexes[p.name] = [p.x_axis[0]]
+
+        parent.save_state("correlation-store-obs-range", store_obs_range)
+        return (
+            {"min-height": "auto", "padding-top": "20px"},
```

Comment:
i think this one might be responsible for the collapse of the observation selector plot when top selectors (ensembles, responses, parameters) get collapsed - it ain't pretty, i don't like hard coding sizes, but i also have a very bad handle on css in general, but setting this to `300px` stopped it from collapsing on me